server/events/project_command_builder.go

```diff
@@ -106,16 +107,28 @@ func (p *DefaultProjectCommandBuilder) buildPlanAllCommands(ctx *CommandContext,
 	ctx.Log.Debug("got workspace lock")
 	defer unlockFn()
 
+	repoDir, _, err := p.WorkingDir.Clone(ctx.Log, ctx.BaseRepo, ctx.HeadRepo, ctx.Pull, workspace)
+	if err != nil {
+		return nil, err
+	}
+
 	// We'll need the list of modified files.
 	modifiedFiles, err := p.VCSClient.GetModifiedFiles(ctx.BaseRepo, ctx.Pull)
 	if err != nil {
 		return nil, err
 	}
 	ctx.Log.Debug("%d files were modified in this pull request", len(modifiedFiles))
 
-	repoDir, _, err := p.WorkingDir.Clone(ctx.Log, ctx.BaseRepo, ctx.HeadRepo, ctx.Pull, workspace)
-	if err != nil {
-		return nil, err
+	// If the --all flag has been given we just suppose all files have been
```

Comment:
I think rather than piggy-backing on `modifiedFiles` it makes sense to add new functions to `ProjectFinder`: `AllProjectsViaConfig` and `AllProjects`. If `all == true` then we'd skip the `GetModifiedFiles` call and then call either of those two methods instead of `DetermineProjectsViaConfig` and `DetermineProjects` (respectively).